File: python/silnia.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# a! = 1 dla {0,1}
# a! = 1*2*...*n N+{0,1}

# n! = (n-1)! * n
# 4! = 3! * 4


def silnia_rek(n):
    if n < 2:
        return 1
    return silnia_rek(n - 1) * n


def silnia_it(n):
    """Funkcja oblicza iteracyjnie potege l. naturalnej"""
    wynik = 1
    for i in range(2, n + 1):
        wynik = wynik * i
    return wynik


def main(args):
    a = int(input("Podaj liczba:"))
    assert type(a) == int

    assert silnia_it(0) == 1
    assert silnia_it(1) == 1
    assert silnia_it(2) == 2
    assert silnia_it(6) == 720

    assert silnia_rek(0) == 1
    assert silnia_rek(1) == 1
    assert silnia_rek(2) == 2
    assert silnia_rek(6) == 720

    print("Wynik iteracyjny: ", silnia_it(a), ' \n ')
    print("Wynik rekurencyjny: ", silnia_rek(a))

    return 0


if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/silnia.cpp
/*
 * potegowanie.cpp
 */

#include <iostream>

using namespace std;

//Deklaracja funkcji
int silnia_it(int n);
int silnia_rek(int n);

int main(int argc, char **argv)
{
    int liczba = 0;
    
    cout<<"Podaj liczbe: ";
    cin>>liczba;
        
    cout<<"Wynik_it: "<<silnia_it(liczba)<<endl;
    cout<<"Wynik_rek: "<<silnia_it(liczba);
    
    return 0;
}

//Funkcje
int silnia_it(int n)
{
    int wynik = 1;
    
    for(int i = 2; i<=n; i++)
    {
        wynik=wynik*i;
    }
    return wynik;
}
int silnia_rek(int n)
{
    if(n<2)
        return 1;
    return silnia_rek(n-1) * n;
}

